### ISSUE TITLE
🐛 BUG: Wrangler dev fails when CF Access is enabled on `*.domain.tld`

### ISSUE THREAD

**Issue by Skye-31** [cloudflare/wrangler2#1906]
### What version of `Wrangler` are you using?

2.1.6

### What operating system are you using?

Windows (don't think it makes a difference here)

### Describe the Bug

1. Create a worker
2. Create a Cloudflare Access application, with routes `*.yourdomain.com` and `yourdomain.com`
3. Run `wrangler dev`, and attempt to visit your worker
4. Find the following error message ![image](https://user-images.githubusercontent.com/46286597/191810684-dbe06f06-91a6-4d92-be73-96be210b0472.png)

(worth noting this is also logged) 
![image](https://user-images.githubusercontent.com/46286597/191811285-2573059b-1757-4015-abc2-e8177f552142.png)

**Comment by cameron-robey** [comment]
Hey 👋 !

I've done some investigation into this, and I have been trying to reproduce this.

The steps I took:

1. Create a worker and add a custom domain to it in `wrangler.toml`
2. Create an access application with that domain
3. Run `npx wrangler dev`

Interestingly I am getting no errors in the console, but I get an `ERROR 1000: DNS points to prohibited IP` in browser.
Everything works ok with wrangler publish.

Firstly, could you provide some further steps for reproduction, or retrying this with a different domain?
Secondly, I think that it could be possible to pass through a token in our proxy server to allow `wrangler dev` to access Cloudflare Access applications - we will discuss this internally.

**Comment by Skye-31** [comment]
Relevant wrangler toml here
I believe this only occurs when the access policy explicitly covers `*.example.com`, as `wrangler dev` will then run on `somesubdomain.example.com`.
```toml
name = "core"
main = "src/index.ts"
compatibility_date = "2022-08-19"
account_id = "snip"

usage_model = "bundled"
d1_databases = [
	{ binding = "DB", database_name = "core", database_id = "snip", preview_database_id = "snip" }
]
routes = [
	{ pattern = "example.com/api*", zone_id = "snip" }
]
```

**Comment by cameron-robey** [comment]
When I was using a wildcard, I was still getting the error that I had before interestingly.

My guess is that these are symptoms of a wider problem of how should `wrangler dev` remote mode should interact with Cloudflare access, which is a story that probably hasn't been fully thought out? 

Will discuss internally and do some investigation to hopefully find a solution :)

**Comment by Skye-31** [comment]
Thanks! 😄 
For now I'm using local mode, but unfortunately no analytics engine there yet 😓

**Comment by Skye-31** [comment]
Yep! Just wrangler dev

**Comment by penalosa** [comment]
@Skye-31 Would you be able to see if this pre-release version of wrangler fixes your issue? `npx https://prerelease-registry.developers.workers.dev/runs/3189869870/npm-package-wrangler-1982 dev`

**Comment by penalosa** [comment]
@Skye-31 Apologies, this was on the back burner for a bit, but it should be updated now. Could you check and see if the PR as it stands now will work for you?

**Comment by willtemperley** [comment]
Tried that, it's running wrangler 0.0.0-67f4089a but seems the same.  Getting 302s still with the same error message. 

Access to fetch at 'https://opencragmap.cloudflareaccess.com/cdn-cgi/access/login/localhost:8887?kid=<redacted> (redirected from 'http://localhost:5173/authenticated/api/boundary?countryId=300') from origin 'http://localhost:5173' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by willtemperley** [comment]
Seems to work now, the only change I made was adding the application to the Zero Trust App Launcher

**Comment by penalosa** [comment]
We released `wrangler@2.2.1` 2 days ago, which contained fixes for `wrangler dev` and Cloudflare Access. Would you be able to check if that's the version that resolved the bug, and if you're still seeing the issues on `wrangler@2.1.5`?
